From 46f4cfe513f0f893c862dca7f02ed7d7932b2115 Mon Sep 17 00:00:00 2001
Message-Id: <46f4cfe513f0f893c862dca7f02ed7d7932b2115@dist-git>
From: Jiri Denemark <jdenemar@redhat.com>
Date: Tue, 26 Apr 2022 12:50:41 +0200
Subject: [PATCH] cpu_map: Disable cpu64-rhel* for host-model and baseline

These ancient RHEL-only CPU models should not really be used by any CPU
definition created by libvirt. We keep them just for backwards
compatibility with domains which might still be using them.

Signed-off-by: Jiri Denemark <jdenemar@redhat.com>
Reviewed-by: Michal Privoznik <mprivozn@redhat.com>
(cherry picked from commit d2e4d66be35cd04da72e5f5129a8a4da6a931505)

https://bugzilla.redhat.com/show_bug.cgi?id=1851227

Signed-off-by: Jiri Denemark <jdenemar@redhat.com>
---
 src/cpu_map/x86_cpu64-rhel5.xml                        | 2 +-
 src/cpu_map/x86_cpu64-rhel6.xml                        | 2 +-
 tests/cputestdata/x86_64-baseline-no-vendor-result.xml | 3 ++-
 3 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/cpu_map/x86_cpu64-rhel5.xml b/src/cpu_map/x86_cpu64-rhel5.xml
index be6bcdb7a6..7402b7603c 100644
--- a/src/cpu_map/x86_cpu64-rhel5.xml
+++ b/src/cpu_map/x86_cpu64-rhel5.xml
@@ -1,6 +1,6 @@
 <cpus>
   <model name='cpu64-rhel5'>
-    <decode host='on' guest='on'/>
+    <decode host='off' guest='off'/>
     <feature name='apic'/>
     <feature name='clflush'/>
     <feature name='cmov'/>
diff --git a/src/cpu_map/x86_cpu64-rhel6.xml b/src/cpu_map/x86_cpu64-rhel6.xml
index c62b1b5575..061939c733 100644
--- a/src/cpu_map/x86_cpu64-rhel6.xml
+++ b/src/cpu_map/x86_cpu64-rhel6.xml
@@ -1,6 +1,6 @@
 <cpus>
   <model name='cpu64-rhel6'>
-    <decode host='on' guest='on'/>
+    <decode host='off' guest='off'/>
     <feature name='apic'/>
     <feature name='clflush'/>
     <feature name='cmov'/>
diff --git a/tests/cputestdata/x86_64-baseline-no-vendor-result.xml b/tests/cputestdata/x86_64-baseline-no-vendor-result.xml
index 00e03b2152..4b4921cf93 100644
--- a/tests/cputestdata/x86_64-baseline-no-vendor-result.xml
+++ b/tests/cputestdata/x86_64-baseline-no-vendor-result.xml
@@ -1,3 +1,4 @@
 <cpu mode='custom' match='exact'>
-  <model fallback='allow'>cpu64-rhel6</model>
+  <model fallback='allow'>kvm64</model>
+  <feature policy='require' name='lahf_lm'/>
 </cpu>
-- 
2.35.1

